### PR TITLE
SPUI-708: Change phone number

### PR DESCRIPTION
**Pull request by DenisMartych**

### CODE REVIEW

File: lib/shared/features/phone_verification/phone_verification_confirm/view/change_phone_verification_confirm.dart [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+                          showResend: false,
+                        );
+                      },
+                    ),
+                  ],
+                ],
+              ),
+            ),
+          ),
+          if (phone.union is Loading) const ScaffoldLoader(),
```
Comment (eli1stark) [inline lib/shared/features/phone_verification/phone_verification_confirm/view/change_phone_verification_confirm.dart:127]: change to SPageFrame loading

**Review by eli1stark — CHANGES_REQUESTED** [review]

File: lib/app/shared/features/profile_details/notifier/change_phone_notifier/change_phone_notifier.dart [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:logging/logging.dart';
+
+import '../../../../../../shared/logging/levels.dart';
+import 'change_phone_state.dart';
+
+class ChangePhoneNotifier extends StateNotifier<ChangePhoneState> {
+  ChangePhoneNotifier({
+    required this.read,
```
Comment (eli1stark) [inline lib/app/shared/features/profile_details/notifier/change_phone_notifier/change_phone_notifier.dart:9]: in this specific case, you can do this without a named parameter, this way it looks cleaner.
`ChangePhoneNotifier(this.read)...`

File: lib/app/shared/features/profile_details/view/components/change_phone_number/change_phone_number.dart [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+import '../../../notifier/change_phone_notifier/change_phone_notipod.dart';
+
+class ChangePhoneNumber extends HookWidget {
+  const ChangePhoneNumber({
+    Key? key,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    final statePhoneNumber = useProvider(phoneNumberNotipod);
```
Comment (eli1stark) [inline lib/app/shared/features/profile_details/view/components/change_phone_number/change_phone_number.dart:23]: phoneNumber

File: lib/app/shared/features/profile_details/view/components/change_phone_number/change_phone_number.dart [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    final notifierPhoneNumber = useProvider(phoneNumberNotipod.notifier);
```
Comment (eli1stark) [inline lib/app/shared/features/profile_details/view/components/change_phone_number/change_phone_number.dart:24]: phoneNumberN

File: lib/app/shared/features/profile_details/view/profile_details.dart [hunk 5]
```diff
@@ -3,16 +3,24 @@ import 'package:flutter_hooks/flutter_hooks.dart';
...
 
 class ProfileDetails extends HookWidget {
   const ProfileDetails({Key? key}) : super(key: key);
 
   @override
   Widget build(BuildContext context) {
-    final authInfo = useProvider(authInfoNotipod);
+    final userInfo = useProvider(userInfoNotipod);
+    final phoneNumberN = useProvider(phoneNumberNotipod.notifier);
+
```
Comment (eli1stark) [inline lib/app/shared/features/profile_details/view/profile_details.dart:22]: remove space

File: lib/app/shared/features/profile_details/view/profile_details.dart [hunk 6]
```diff
@@ -3,16 +3,24 @@ import 'package:flutter_hooks/flutter_hooks.dart';
...
+    final changePhone = useProvider(changePhoneNotipod);
```
Comment (eli1stark) [inline lib/app/shared/features/profile_details/view/profile_details.dart:23]: changePhoneN

File: lib/shared/features/phone_verification/phone_verification_confirm/view/phone_verification_confirm.dart [hunk 7]
```diff
@@ -21,15 +21,22 @@ class PhoneVerificationConfirm extends HookWidget {
   const PhoneVerificationConfirm({
     Key? key,
     required this.onVerified,
+    required this.isChangeFonAlert,
   }) : super(key: key);
 
   final Function() onVerified;
+  final bool isChangeFonAlert;
```
Comment (eli1stark) [inline lib/shared/features/phone_verification/phone_verification_confirm/view/phone_verification_confirm.dart:28]: Fon?

File: lib/shared/features/phone_verification/phone_verification_confirm/view/phone_verification_confirm.dart [hunk 8]
```diff
@@ -21,15 +21,22 @@ class PhoneVerificationConfirm extends HookWidget {
...
 
-  static void push(BuildContext context, Function() onVerified) {
+  static void push(
+    BuildContext context,
+    Function() onVerified, {
```
Comment (eli1stark) [inline lib/shared/features/phone_verification/phone_verification_confirm/view/phone_verification_confirm.dart:32]: use named here

File: lib/shared/features/phone_verification/phone_verification_confirm/view/phone_verification_confirm.dart [hunk 9]
```diff
@@ -21,15 +21,22 @@ class PhoneVerificationConfirm extends HookWidget {
...
     required this.onVerified,
+    required this.isChangeFonAlert,
   }) : super(key: key);
 
   final Function() onVerified;
+  final bool isChangeFonAlert;
 
-  static void push(BuildContext context, Function() onVerified) {
+  static void push(
+    BuildContext context,
```
Comment (eli1stark) [inline lib/shared/features/phone_verification/phone_verification_confirm/view/phone_verification_confirm.dart:31]: use named here

File: lib/shared/notifiers/user_info_notifier/user_info_notifier.dart [hunk 10]
```diff
@@ -154,4 +154,10 @@ class UserInfoNotifier extends StateNotifier<UserInfoState> {
       phone: '',
     );
   }
+
+  void updatePhoneNumber(String phone) {
```
Comment (eli1stark) [inline lib/shared/notifiers/user_info_notifier/user_info_notifier.dart:158]: remove

File: lib/shared/notifiers/user_info_notifier/user_info_state.dart [hunk 11]
```diff
@@ -24,4 +24,6 @@ class UserInfoState with _$UserInfoState {
   const UserInfoState._();
 
   bool get pinEnabled => pin != null;
+
+  bool get enableChangePhoneNumber => phone.isNotEmpty;
```
Comment (eli1stark) [inline lib/shared/notifiers/user_info_notifier/user_info_state.dart:28]: reaname to isPhoneNumberActive

**Review by eli1stark — CHANGES_REQUESTED** [review]

File: lib/app/shared/features/profile_details/notifier/change_phone_notifier/change_phone_notifier.dart (hunk 2 above)
Comment (DenisMartych) [reply lib/app/shared/features/profile_details/notifier/change_phone_notifier/change_phone_notifier.dart:9]: done

File: lib/app/shared/features/profile_details/view/components/change_phone_number/change_phone_number.dart (hunk 3 above)
Comment (DenisMartych) [reply lib/app/shared/features/profile_details/view/components/change_phone_number/change_phone_number.dart:23]: done

File: lib/app/shared/features/profile_details/view/components/change_phone_number/change_phone_number.dart (hunk 4 above)
Comment (DenisMartych) [reply lib/app/shared/features/profile_details/view/components/change_phone_number/change_phone_number.dart:24]: done

File: lib/app/shared/features/profile_details/view/profile_details.dart (hunk 5 above)
Comment (DenisMartych) [reply lib/app/shared/features/profile_details/view/profile_details.dart:22]: done

File: lib/app/shared/features/profile_details/view/profile_details.dart (hunk 6 above)
Comment (DenisMartych) [reply lib/app/shared/features/profile_details/view/profile_details.dart:23]: not used, remove

File: lib/shared/notifiers/user_info_notifier/user_info_notifier.dart (hunk 10 above)
Comment (DenisMartych) [reply lib/shared/notifiers/user_info_notifier/user_info_notifier.dart:158]: remove

File: lib/shared/features/phone_verification/phone_verification_confirm/view/change_phone_verification_confirm.dart (hunk 1 above)
Comment (DenisMartych) [reply lib/shared/features/phone_verification/phone_verification_confirm/view/change_phone_verification_confirm.dart]: done

File: lib/shared/features/phone_verification/phone_verification_confirm/view/phone_verification_confirm.dart (hunk 9 above)
Comment (DenisMartych) [reply lib/shared/features/phone_verification/phone_verification_confirm/view/phone_verification_confirm.dart:29]: done

File: lib/shared/notifiers/user_info_notifier/user_info_state.dart (hunk 11 above)
Comment (DenisMartych) [reply lib/shared/notifiers/user_info_notifier/user_info_state.dart:28]: done

File: lib/app/shared/features/profile_details/view/components/change_password/change_password.dart [hunk 12]
```diff
@@ -30,7 +31,8 @@ class ChangePassword extends HookWidget {
           error: (error) {
             oldPasswordError.value.enableError();
           },
-          orElse: (){},
+          orElse: () {
+          },
```
Comment (eli1stark) [inline lib/app/shared/features/profile_details/view/components/change_password/change_password.dart:35]: remove additional line

**Review by eli1stark — CHANGES_REQUESTED** [review]

File: lib/app/shared/features/profile_details/view/components/change_password/change_password.dart (hunk 12 above)
Comment (DenisMartych) [reply lib/app/shared/features/profile_details/view/components/change_password/change_password.dart:35]: done

File: lib/shared/features/phone_verification/phone_verification_confirm/view/phone_verification_confirm.dart [hunk 13]
```diff
@@ -108,7 +137,7 @@ class PhoneVerificationConfirm extends HookWidget {
               ),
             ),
           ),
-          if (phone.union is Loading) const ScaffoldLoader(),
+          //if (phone.union is Loading) const ScaffoldLoader(),
```
[the remark this answers is not in the document]
Comment (DenisMartych) [reply lib/shared/features/phone_verification/phone_verification_confirm/view/phone_verification_confirm.dart:140]: done

**Pull request merged**